test/common/config/subscription_test_harness.h

```diff
@@ -49,6 +49,10 @@ class SubscriptionTestHarness {
   virtual void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                                    const std::string& version, bool accept) PURE;
 
+  virtual void deliverConfigUpdateWithCode(const std::vector<std::string>& cluster_names,
```

Comment:
Seems like it is better to have an override, i.e.

```
virtual void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                                   const std::string& version, bool accept, bool modify,
                                   const std::string& response_code) PURE;
```

I'm suggesting to have `modify` param here since you will not modify anything if you send `304` (the `onConfigUpdate` callback won't be called).

Reply:
Doesn't changing this signature break things in a bunch of other places? I don't understand how it works in C++ but aren't there functions based off this 'prototype' which need to have the same signature?

Reply:
I meant, you can have two of it: (in `subscription_test_harness.h`)

```c++
virtual void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                                   const std::string& version, bool accept) PURE;

// and...
virtual void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                                   const std::string& version, bool accept, bool modify,
                                   const std::string& response_code) PURE;
```

while in the implementation side (in `http_subscription_test_harness.h`)

```c++
// for the first one:
virtual void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                                   const std::string& version, bool accept) {
	deliverConfigUpdate(cluster_names, version, true, true, "200");
}

// And the second one:
void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                           const std::string& version, bool accept, bool modify,
                           const std::string& response_code) override {
    std::string response_json = "{\"version_info\":\"" + version + "\",\"resources\":[";
    for (const auto& cluster : cluster_names) {
      response_json += "{\"@type\":\"type.googleapis.com/"
                       "envoy.api.v2.ClusterLoadAssignment\",\"cluster_name\":\"" +
                       cluster + "\"},";
    }
    response_json.pop_back();
    response_json += "]}";
    envoy::api::v2::DiscoveryResponse response_pb;
    MessageUtil::loadFromJson(response_json, response_pb);
    Http::HeaderMapPtr response_headers{new Http::TestHeaderMapImpl{{":status", response_code}}};
    Http::MessagePtr message{new Http::ResponseMessageImpl(std::move(response_headers))};
    message->body() = std::make_unique<Buffer::OwnedImpl>(response_json);
    if (modify) {
      EXPECT_CALL(callbacks_, onConfigUpdate(RepeatedProtoEq(response_pb.resources()), version))
          .WillOnce(ThrowOnRejectedConfig(accept));
    }
    if (!accept) {
      EXPECT_CALL(callbacks_, onConfigUpdateFailed(_));
    }
    EXPECT_CALL(random_gen_, random()).WillOnce(Return(0));
    EXPECT_CALL(*timer_, enableTimer(_));
    http_callbacks_->onSuccess(std::move(message));
    if (accept) {
      version_ = version;
    }
    request_in_progress_ = false;
    timerTick();
}
```

Reply:
I pushed what I think were your suggestions, but I can't understand the errors produced during the build